stackdriverlogging.go

```diff
@@ -0,0 +1,196 @@
+func (sl *StackdriverLogging) AddLabel(key, value string) {
+	sl.mtx.Lock()
+	sl.commonLabels[key] = value
+	sl.mtx.Unlock()
+}
+
+// RemoveLabel will remove a common label used to link relevant log entries together
+func (sl *StackdriverLogging) RemoveLabel(key string) {
+	sl.mtx.Lock()
+	delete(sl.commonLabels, key)
+	sl.mtx.Unlock()
+}
+
+// Debugf will log the data in the specified format with a severity level of debug
+func (sl *StackdriverLogging) Debugf(format string, args ...interface{}) {
+	sl.formattedStringLog(logging.Debug, format, args...)
+}
+
+// Infof will log the data in the specified format with a severity level of info
+func (sl *StackdriverLogging) Infof(format string, args ...interface{}) {
+	sl.formattedStringLog(logging.Info, format, args...)
+}
+
+// Warningf will log the data in the specified format with a severity level of warning
+func (sl *StackdriverLogging) Warningf(format string, args ...interface{}) {
+	sl.formattedStringLog(logging.Warning, format, args...)
+}
+
+// Errorf will log the data in the specified format with a severity level of error
+func (sl *StackdriverLogging) Errorf(format string, args ...interface{}) {
+	sl.formattedStringLog(logging.Error, format, args...)
+}
+
+// Criticalf will log the data in the specified format with a severity level of critical
+func (sl *StackdriverLogging) Criticalf(format string, args ...interface{}) {
+	sl.formattedStringLog(logging.Critical, format, args...)
+}
+
+func (sl *StackdriverLogging) Request(method, url, format string, args ...interface{}) {}
+
+// Debug will log the specified data to the logging system with the debug log level
+func (sl *StackdriverLogging) Debug(data interface{}) {
+	sl.processLog(logging.Debug, data)
+}
+
+// Info will log the specified data to the logging system with the info log level
+func (sl *StackdriverLogging) Info(data interface{}) {
+	sl.processLog(logging.Info, data)
+}
+
+// Warning will log the specified data to the logging system with the warning log level
+func (sl *StackdriverLogging) Warning(data interface{}) {
+	sl.processLog(logging.Warning, data)
+}
+
+// Error will log the specified data to the logging system with the error log level
+func (sl *StackdriverLogging) Error(data interface{}) {
+	sl.processLog(logging.Error, data)
+}
+
+// Critical will log the specified data to the logging system with the critical log level
+func (sl *StackdriverLogging) Critical(data interface{}) {
+	sl.processLog(logging.Critical, data)
+}
+
+// formattedStringLog will format the string and log it with the child logger at the applicable log level
+func (sl *StackdriverLogging) formattedStringLog(severity logging.Severity, format string, args ...interface{}) {
+	msg := fmt.Sprintf(format, args...)
+	sl.processLog(severity, msg)
+}
+
+// processLog will create a new log entry with the child logger
+func (sl *StackdriverLogging) processLog(severity logging.Severity, data interface{}) {
```

Comment:
Why can't this call the underlying log interface directly?  It would greatly simplify the code to not have to pass these messages over a channel.